bc_obps/service/data_access_service/document_service_v2.py

```diff
@@ -0,0 +1,31 @@
+from typing import Optional
+from uuid import UUID
+from service.data_access_service.operation_service import OperationDataAccessService
+from registration.models import Document, DocumentType
+from django.core.files.base import ContentFile
+
+
+class DocumentDataAccessServiceV2:
+    @classmethod
+    def get_operation_document_by_type(cls, operation_id: UUID, document_type: str) -> Document | None:
```

Comment:
I got lost trying to follow the paths of where this function gets called, so apologies if this question is wrong, but do we need to add a check here based on user_guid to confirm that the user has permission to access data for the operation in question? (I couldn't tell if this gets checked somewhere else)

Reply:
I made a get_if_authorized function to this, that's how we've generally done it in other services. Thanks for catching this!